css-mixins-1/Overview.bs

```diff
@@ -0,0 +1,538 @@
+rather than the computed-value based substitution
+described in [[!css-variables]].
+
+<div algorithm>
+	To <dfn>substitute a dashed function</dfn> in a value,
+		with |dashed function| being a <<dashed-function>>:
+
+		1. Let |function| be the result of dereferencing
+			the |dashed function|'s name as a [=tree-scoped reference=].
+			If no such name exists, return failure.
+		2. Let |dependency values| be an initially empty [=list=].
+		2. For each |dependency| in |function|'s [=function dependency|dependencies=]:
+			* Let |dependency value| be the value that would be substituted
+				if a ''var()'' function had been specified explicitly
+				at the end of |dashed function|'s argument list,
+				with |dependency| as its only argument.
+			* If that substitution would have made a containing declaration
+				[=invalid at computed-value time=],
+				set |dependency value| to the [=guaranteed-invalid value=].
+			* If |dependency value| is the [=guaranteed-invalid value=],
+				and |dependency| has a [=default value=],
+				set |dependency value| to that [=default value=].
+			* If |dependency| has a [=parameter type|type=],
+				set |dependency value| to the result of
+				[=resolve a typed value|resolving a typed value=],
+				using |dependency value| as the value,
+				and the [=syntax definition=] associated with the dependency's type as the syntax.
+			* Append |dependency value| to |dependency values|.
+		1. [=Evaluate a custom function=],
+			using |function|, |dashed function| and |dependency values|.
+		2. If failure was returned, return failure.
+		3. Otherwise,
+			replace the <<dashed-function>> with the [=equivalent token sequence=]
+			of the value resulting from the evaluation.
+</div>
+
+If [=substitute a dashed function=] fails,
+and the substitution is taking place on a property's value,
+then the declaration containing the <<dashed-function>> becomes
+[=invalid at computed-value time=].
+
+Evaluating Custom Functions {#evaluating-custom-functions}
+----------------------------------------------------------
+
+<div algorithm>
+	To <dfn>evaluate a custom function</dfn>,
+	with |function| being a [=custom function=],
+	|dashed function| being the <<dashed-function>> invoking that |function|,
+	and |dependency values| being a [=list=] of values.
+
+	1. If the number of values in |dashed function|'s argument list
+		is greater than the number of values in |function|'s [=function parameter|parameters=],
+		return failure.
+	2. For each value |parameter| in |function|'s [=function parameter|parameters=],
+		let |argument| be the corresponding value in |dashed function|'s argument list
+		at the same index:
+		* If |argument| does not exist,
+			set |argument| to the [=guaranteed-invalid value=].
+		* If |argument| is the [=guaranteed-invalid value=],
+			and |parameter| has a [=default value=],
+			set |argument| to that [=default value=].
+		* If |parameter| has a [=parameter type|type=],
```

Comment:
This just turns the passed value into the guaranteed-invalid value if it doesn't match the type. If there's a default value, do we want to resolve it to that instead? (I'm really not sure.)

Reply:
I _think_ what you suggest is more consistent, since not existing at all (which is where `<guaranteed-invalid>` came from in the first place) gives you the default value.

Although it's possible to argue both ways. If we take the default value, then it's like a function-wide fallback, and we don't need to duplicate the same fallback (potentially) in every `var()`. However, then we also can't use fallback A in one place, and fallback B in another place. The solution to this might be that authors simply _don't_ provide a default when they want per-`var()` fallbacks.

Reply:
Done.